app/lib/features/chat/providers/notifiers/media_chat_notifier.dart

```diff
@@ -0,0 +1,72 @@
+import 'dart:io';
+import 'package:acter/common/providers/chat_providers.dart';
+import 'package:acter/features/chat/models/media_chat_state/media_chat_state.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:path_provider/path_provider.dart';
+
+class MediaChatNotifier extends StateNotifier<MediaChatState> {
+  final Ref ref;
+  final String messageId;
+  Convo? _convo;
+
+  MediaChatNotifier({
+    required this.messageId,
+    required this.ref,
+  }) : super(const MediaChatState()) {
+    _init();
+  }
+
+  void _init() async {
+    _convo = await ref.read(currentConvoProvider.future);
```

Comment:
@gnunicorn 
Yes, understood the issue and how it can be dangerous. 
Currently, adding `FIXME` statements here but I will figure it out on a later PR and remove the `currentConvoProvider`